driver/src/main/java/org/neo4j/driver/internal/security/SecurityPlanImpl.java

```diff
@@ -38,54 +40,112 @@
+    public static SecurityPlan forCustomCASignedCertificates( File certFile, boolean requiresHostnameVerification,
+                                                              boolean requiresRevocationChecking )
             throws GeneralSecurityException, IOException
     {
-        // A certificate file is specified so we will load the certificates in the file
-        // Init a in memory TrustedKeyStore
-        KeyStore trustedKeyStore = KeyStore.getInstance( "JKS" );
+        SSLContext sslContext = configureSSLContext( certFile, requiresRevocationChecking );
+        return new SecurityPlanImpl( true, sslContext, requiresHostnameVerification, requiresRevocationChecking );
+    }
+
+    public static SecurityPlan forSystemCASignedCertificates( boolean requiresHostnameVerification, boolean requiresRevocationChecking )
+            throws GeneralSecurityException, IOException
+    {
+        SSLContext sslContext = configureSSLContext( null, requiresRevocationChecking );
+        return new SecurityPlanImpl( true, sslContext, requiresHostnameVerification, requiresRevocationChecking );
+    }
+
+    public static SSLContext configureSSLContext( File customCertFile, boolean requiresRevocationChecking )
+            throws GeneralSecurityException, IOException
+    {
+        KeyStore trustedKeyStore = KeyStore.getInstance( KeyStore.getDefaultType() );
         trustedKeyStore.load( null, null );
 
-        // Load the certs from the file
-        loadX509Cert( certFile, trustedKeyStore );
+        if ( customCertFile != null )
+        {
+            // A certificate file is specified so we will load the certificates in the file
+            loadX509Cert( customCertFile, trustedKeyStore );
+        }
+        else
+        {
+            loadSystemCertificates( trustedKeyStore );
+        }
+
+        // Configure certificate revocation checking (X509CertSelector() selects all certificates)
+        PKIXBuilderParameters pkixBuilderParameters = new PKIXBuilderParameters( trustedKeyStore, new X509CertSelector() );
+
+        // sets checking of stapled ocsp response
+        pkixBuilderParameters.setRevocationEnabled( requiresRevocationChecking );
+
+        // enables status_request exentension in client hello
+        if ( requiresRevocationChecking )
+        {
+            System.setProperty( "jdk.tls.client.enableStatusRequestExtension", "true" );
+        }
 
         // Create TrustManager from TrustedKeyStore
-        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance( "SunX509" );
-        trustManagerFactory.init( trustedKeyStore );
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance( TrustManagerFactory.getDefaultAlgorithm() );
+        trustManagerFactory.init( new CertPathTrustManagerParameters( pkixBuilderParameters ) );
 
         SSLContext sslContext = SSLContext.getInstance( "TLS" );
+
```

Comment:
Maybe it's just me, but I do understand the code a bit better when we bring the two steps a bit closer together.

```suggestion
        SSLContext sslContext = SSLContext.getInstance( "TLS" );

        // Create TrustManager from TrustedKeyStore and use them to initialize the above context
        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance( TrustManagerFactory.getDefaultAlgorithm() );
        trustManagerFactory.init( new CertPathTrustManagerParameters( pkixBuilderParameters ) );
        sslContext.init( new KeyManager[0], trustManagerFactory.getTrustManagers(), null );
```